### PR TITLE
builder-next: update to new buildkit and fix upgrade bugs

### LINKED ISSUE #41077: buildkit build cache migration  (v19.03.8 -> master)
**Description**

We upgrade nodes from Docker Version 19.03.08 to master branch for new features of buidkit, but few builds failed with :
> failed to prepare ... : open ... : no such file or directory

and we finnally dug out that the reason maybe the buildkit cache of 19.03 is not compliance with the latest version, we can clear the path `/var/lib/docker/buildkit` to avoid the failure, but we'd have better choice when it released in the coming 20.03 version.

**Steps to reproduce the issue:**
1.  build with 19.03.08
2.  upgrade version to master branch
3.  build with the same base image, it wil be failed

**Describe the results you received:**
```
$ docker info
Client:
 Debug Mode: false

Server:
 Containers: 1
  Running: 0
  Paused: 0
  Stopped: 1
 Images: 42
 Server Version: 19.03.8
 Storage Driver: overlay2
  Backing Filesystem: <unknown>
  Supports d_type: true
  Native Overlay Diff: true
 Logging Driver: json-file
 Cgroup Driver: cgroupfs
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
 Swarm: inactive
 Runtimes: runc
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: 7ad184331fa3e55e52b890ea95e65ba581ae3429
 runc version: dc9208a3303feef5b3839f4323d9beb36df0a9dd
 init version: fec3683
 Security Options:
  seccomp
   Profile: default
 Kernel Version: 3.10.0-693.2.2.el7.x86_64
 Operating System: CentOS Linux 7 (Core)
 OSType: linux
 Architecture: x86_64
 CPUs: 1
 Total Memory: 1.796GiB
 Name: zcm-test
 ID: ZAGZ:H2UI:BEYE:R7GA:G2K2:ZIPS:VWDQ:MHS3:DVB6:KD4K:4BQB:NA2C
 Docker Root Dir: /var/lib/docker
 Debug Mode: false
 Registry: https://index.docker.io/v1/
 Labels:
 Experimental: true
 Insecure Registries:
  127.0.0.0/8
 Live Restore Enabled: false

$ ls -trl
total 16
-rw-r--r-- 1 root root  4 Jun  7 15:22 foo
-rw-r--r-- 1 root root  6 Jun  7 15:22 bar
-rw-r--r-- 1 root root 26 Jun  7 15:25 Dockerfile_foo
-rw-r--r-- 1 root root 26 Jun  7 15:25 Dockerfile_bar
$  cat foo
foo
$  cat bar
bar
$  cat Dockerfile_foo 
FROM busybox 
COPY foo /
$  cat Dockerfile_bar
FROM busybox 
COPY bar /

$ export DOCKER_BUILDKIT=1
$ docker build -t foo -f Dockerfile_foo .
[+] Building 3.9s (7/7) FINISHED                                                                                      
 => [internal] load build definition from Dockerfile_foo                                                         0.0s
 => => transferring dockerfile: 67B                                                                              0.0s
 => [internal] load .dockerignore                                                                                0.1s
 => => transferring context: 2B                                                                                  0.0s
 => [internal] load metadata for docker.io/library/busybox:latest                                                3.0s
 => [internal] load build context                                                                                0.0s
 => => transferring context: 34B                                                                                 0.0s
 => [1/2] FROM docker.io/library/busybox@sha256:95cf004f559831017cdf4628aaf1bb30133677be8702a8c5f2994629f637a20  0.8s
 => => resolve docker.io/library/busybox@sha256:95cf004f559831017cdf4628aaf1bb30133677be8702a8c5f2994629f637a20  0.0s
 ...
 => => extracting sha256:76df9210b28cbd4bc127844914d0a23937ed213048dc6289b2a2d4f7d675c75e                        0.1s
 => [2/2] COPY foo /                                                                                             0.0s
 => exporting to image                                                                                           0.0s
 => => exporting layers                                                                                          0.0s
 => => writing image sha256:87722e065e0892f1a03c26a9a699a035a3cff59c3c92c83d43f082d1d51438a7                     0.0s
 => => naming to docker.io/library/foo  

$ systemctl stop docker
$ cp ~/dockerd-dev /usr/bin/dockerd
$ systemctl start docker

$  docker version
Server:
 Engine:
  Version:          dev
  API version:      1.41 (minimum version 1.12)
  Go version:       go1.13.8
  Git commit:       5ffd677824
  Built:            Tue Jun  2 19:57:05 2020
  OS/Arch:          linux/amd64
  Experimental:     true
 containerd:
  Version:          1.2.13
  GitCommit:        7ad184331fa3e55e52b890ea95e65ba581ae3429
 runc:
  Version:          1.0.0-rc10
  GitCommit:        dc9208a3303feef5b3839f4323d9beb36df0a9dd
 docker-init:
  Version:          0.18.0
  GitCommit:        fec3683

$  docker build -t bar -f Dockerfile_bar .
[+] Building 3.0s (6/6) FINISHED                                                                                      
 => [internal] load build definition from Dockerfile_bar                                                         0.0s
 => => transferring dockerfile: 67B                                                                              0.0s
 => [internal] load .dockerignore                                                                                0.0s
 => => transferring context: 2B                                                                                  0.0s
 => [internal] load metadata for docker.io/library/busybox:latest                                                2.9s
 => [internal] load build context                                                                                0.0s
 => => transferring context: 36B                                                                                 0.0s
 => CACHED [1/2] FROM docker.io/library/busybox@sha256:95cf004f559831017cdf4628aaf1bb30133677be8702a8c5f2994629  0.0s
 => ERROR [2/2] COPY bar /                                                                                       0.0s
------
 > [2/2] COPY bar /:
------
failed to prepare ichmd73l4mnzof7inzjponubi: open /var/lib/docker/overlay2/sha256:1be74353c3d0fd55fb5638a52953e6f1bc441e5b1710921db9ec2aa202725569/committed: no such file or directory

$ docker system df
Error response from daemon: error getting build cache usage: failed to get usage for sok2lp7lqrmpmh7d09eep3kxi: lstat /var/lib/docker/overlay2/sha256:1be74353c3d0fd55fb5638a52953e6f1bc441e5b1710921db9ec2aa202725569/diff: no such file or directory
$ docker system prune
WARNING! This will remove:
  - all stopped containers
  - all networks not used by at least one container
  - all dangling images
  - all dangling build cache

Are you sure you want to continue? [y/N] y
Error response from daemon: failed to prune build cache: failed to remove a48fl9rx4d4ksq0x47v9wedl0: lease "a48fl9rx4d4ksq0x47v9wedl0": not found
```

**Describe the results you expected:**
build success

### PR DESCRIPTION
**Pull request by tonistiigi**
fixes https://github.com/moby/moby/issues/41077

Upgrades buildkit to bring in buildkit side fix moby/buildkit#1529

### CODE REVIEW

File: builder/builder-next/controller.go [hunk 1]
```diff
@@ -86,7 +86,11 @@ func newController(rt http.RoundTripper, opt Opt) (*control.Controller, error) {
 		return nil, err
 	}
 
-	md, err := metadata.NewStore(filepath.Join(root, "metadata.db"))
+	if err := cache.MigrateV2(context.TODO(), filepath.Join(root, "metadata.db"), filepath.Join(root, "metadata_v2.db"), store, snapshotter, lm); err != nil {
```
Comment (AkihiroSuda) [inline builder/builder-next/controller.go:89]: nit: `context.Background()`?

**Review by AkihiroSuda — APPROVED** [review]

File: builder/dockerfile/dispatchers.go [hunk 2]
```diff
@@ -205,7 +205,8 @@ func dispatchTriggeredOnBuild(d dispatchRequest, triggers []string) error {
 		}
 		cmd, err := instructions.ParseCommand(ast.AST.Children[0])
 		if err != nil {
-			if instructions.IsUnknownInstruction(err) {
```
Comment (thaJeztah) [inline builder/dockerfile/dispatchers.go:208]: Looks like this was removed in the vendored version; should this be in the same commit to not break git bisect?

Comment (tonistiigi) [reply builder/dockerfile/dispatchers.go:208]: I guess this is what `git bisect skip` is for. I can put everything to a single commit if you like. Though this is easier to review.

**Comment by tonistiigi** [comment]
Seems some unit tests are broken. Checking.

File: builder/dockerfile/dispatchers.go (hunk 2 above)
Comment (thaJeztah) [reply builder/dockerfile/dispatchers.go:208]: With github hiding vendored files, it's probably ok to squash the commits.

**Comment by AkihiroSuda** [comment]
ping @thaJeztah @tiborvass

**Review by thaJeztah — APPROVED** [review]
oops, thought I approved already

LGTM

**Pull request merged**